GeoHealthCheck/models.py

```diff
@@ -497,6 +497,62 @@ def db_commit():
         print(msg)
 
 
+# complete handle of resource test
+def run_resource(resourceid):
+    resource = Resource.query.filter_by(identifier=resourceid).first()
+
+    APP = App.get_app()
+    from healthcheck import run_test_resource
+
+    if not resource.active:
+        # Exit test of resource if it's not active
+        return
+
+    # Get the status of the last run,
+    # assume success if there is none
+    last_run_success = True
+    last_run = resource.last_run
+    if last_run:
+        last_run_success = last_run.success
+
+    # Run test
+    result = run_test_resource(resource)
+
+    run1 = Run(resource, result, datetime.utcnow())
+
+    DB.session.add(run1)
+
+    # commit or rollback each run to avoid long-lived transactions
+    # see https://github.com/geopython/GeoHealthCheck/issues/14
+    db_commit()
+
+    if APP.config['GHC_NOTIFICATIONS']:
+        # Attempt notification
+        try:
+            notify(APP.config, resource, run1, last_run_success)
+        except Exception as err:
+            # Don't bail out on failure in order to commit the Run
+            msg = str(err)
+            print('error notifying: %s' % msg)
+    if not __name__ == '__main__':
+        DB.session.remove()
+
+
+# Complete handle of old runs deletion
+def flush_runs():
+    APP = App.get_app()
+    retention_time = timedelta(*APP.config['GHC_RETENTION_DAYS'])
```

Comment:
See earlier comments: may just calculate retention time from days, e.g. `retention_time ` in milliseconds will be something like: 
`GHC_RETENTION_DAYS * 24 * 3600 * 1000`